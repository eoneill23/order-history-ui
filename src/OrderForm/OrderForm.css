.input {
  width: 80%;
}

.form-btn {
  background-color: green;
  border: none;
  height: 30px;
  width: 80%;
}