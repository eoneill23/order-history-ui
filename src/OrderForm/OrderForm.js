import React, { Component } from 'react';

class OrderForm extends Component {
  constructor() {
    super();
    this.state = {
      img: '',
      name: '',
      description: '',
      price: ''
    }
  }

  handleInputChange = (e) => {
    this.setState({[e.target.name]: e.target.value})
  }

  createOrder = (e) => {
    e.preventDefault();
    let newOrder = {
      id: Date.now(),
      ...this.state
    }
    this.props.addOrder(newOrder);
    this.clearInputs();
  }

  clearInputs = () => {
    this.setState({
      img: '',
      name: '',
      description: '',
      price: ''
    })
  }

  render() {
    return (
      <form className='purchase-form'>
        <input 
          type='text'
          name='img'
          value={this.state.img}
          placeholder='Enter an image URL'
          onChange={this.handleInputChange}
        />
        <input
          type='text'
          name='name'
          value={this.state.name}
          placeholder='Enter an order name'
          onChange={this.handleInputChange}
        />
        <input
          type='text'
          name='description'
          value={this.state.description}
          placeholder='Enter an order description'
          onChange={this.handleInputChange}
        />
        <input
          type='text'
          name='price'
          value={this.state.price}
          placeholder='Enter an order price'
          onChange={this.handleInputChange}
        />
        <button onClick={(e) => this.createOrder(e)}>
          Add Purchase
        </button>
      </form>
    )
  }
}

export default OrderForm;