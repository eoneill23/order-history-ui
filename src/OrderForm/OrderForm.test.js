import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import OrderForm from './OrderForm.js';

describe('OrderForm', () => {
  let wrapper, addOrderMock;

  beforeEach(() => {
    addOrderMock = jest.fn();
    wrapper = shallow(<OrderForm addOrder={addOrderMock}/>)
  });

  it('should render without crashing', () => {
    expect(wrapper.length).toEqual(1)
  });

  it('should match the snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should update state when change is detected in the img input', () => {
    const e = { target: { name: 'img', value: 'twitter.com' } };

    wrapper.find('input').at(0).simulate('change', e);

    expect(wrapper.state('img')).toEqual('twitter.com')
  });

  it('should update state when change is detected in the name input', () => {
    const e = {target: {name: 'name', value: 'bike'}};

    wrapper.find('input').at(0).simulate('change', e);

    expect(wrapper.state('name')).toEqual('bike')
  });

  it('should update state when change is detected in the description input', () => {
    const e = { target: { name: 'description', value: 'fast bike' } };

    wrapper.find('input').at(0).simulate('change', e);

    expect(wrapper.state('description')).toEqual('fast bike')
  });

  it('should update state when change is detected in the price input', () => {
    const e = { target: { name: 'price', value: '200' } };

    wrapper.find('input').at(0).simulate('change', e);

    expect(wrapper.state('price')).toEqual('200')
  });

  it('should fire the createOrder method when the button is clicked', () => {
    const e = {preventDefault: jest.fn()};

    wrapper.find('button').simulate('click', e);

    expect(addOrderMock).toHaveBeenCalled();
  });

  it('should fire the clearInputs method when the button is clicked', () => {
    const e = { preventDefault: jest.fn() };
    wrapper.instance().clearInputs = jest.fn()

    wrapper.find('button').simulate('click', e);

    expect(wrapper.instance().clearInputs).toHaveBeenCalled();
  });
});