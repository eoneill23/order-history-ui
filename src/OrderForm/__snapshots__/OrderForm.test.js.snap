// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderForm should match the snapshot with all data passed in correctly 1`] = `
<form
  className="purchase-form"
>
  <input
    className="input"
    name="img"
    onChange={[Function]}
    placeholder="Enter an image URL"
    type="text"
    value=""
  />
  <input
    className="input"
    name="name"
    onChange={[Function]}
    placeholder="Enter an order name"
    type="text"
    value=""
  />
  <input
    className="input"
    name="description"
    onChange={[Function]}
    placeholder="Enter an order description"
    type="text"
    value=""
  />
  <input
    className="input"
    name="price"
    onChange={[Function]}
    placeholder="Enter an order price"
    type="text"
    value=""
  />
  <button
    className="form-btn"
    onClick={[Function]}
  >
    Add Purchase
  </button>
</form>
`;
